### ISSUE TITLE
Kernel panic while installing on Debian 10

### ISSUE THREAD

**Issue by gonzague** [pi-hole/pi-hole#3498]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate
**How familiar are you with the the source code relevant to this issue?:**

`2`

---
**Expected behaviour:**

Installation of Pi-Hole on a brand new Debian 10 VM 

**Actual behaviour:**

Installation stops with the machine doing a kernel panic just after trying to install pihole-FTL 
`+ url=https://github.com/pi-hole/ftl/releases/latest/download
+ curl -sSL --fail https://github.com/pi-hole/ftl/releases/latest/download/pihole-FTL-linux-x86_64 -o pihole-FTL-linux-x86_64
+ curl -sSL --fail https://github.com/pi-hole/ftl/releases/latest/download/pihole-FTL-linux-x86_64.sha1 -o pihole-FTL-linux-x86_64.sha1

Message from syslogd@antipub at Jun 16 14:51:38 ...
 kernel:[  340.380646] Kernel panic - not syncing: Fatal exception in interrupt`

**Steps to reproduce:**

Install a Debian 10 VM from the netinst ISO ( https://saimei.ftp.acc.umu.se/debian-cd/current/amd64/iso-cd/debian-10.4.0-amd64-netinst.iso ) on a VMWare ESXi 6.5 server 
Launch the install of Pi-Hole using the command `curl -sSL https://install.pi-hole.net 3 | sudo bash -x`
**Troubleshooting undertaken, and/or other relevant information:**

I have restarted the VM and retried several times to install Pi-Hole, without success. I don't have specific settings. I tried the "Clone our repository and run" option as well

**Comment by gonzague** [comment]
I found the "culprit" if it is of interest, using the VMX3 Ethernet card on the VM = crash 
Using the E1000e Ethernet card works

Now I'm getting different errors in the install process such as :
```
_main: line 1892: usermod: command not found
grep: /etc/pihole/setupVars.conf: No such file or directory_
```
and their friend `main: line 1758: /opt/pihole/gravity.sh: No such file or directory`

**Comment by dschaper** [comment]
`usermod` is a system utility that is needed. If it's missing from the netinst ISO then you'll need an image that has the basic utilities.

**Comment by gonzague** [comment]
hey @dschaper weirdly the netinst ISO is supposed to download these basic utilities as I ticked _standard utilities_ when I installed it 

and my system has this utility.. I just need to use sudo to be able to use it:

```
root@antipub:/home/gonzague# sudo usermod -a -G sudo user
usermod: user 'user' does not exist
```
```
root@antipub:/home/gonzague# usermod
bash: usermod: command not found
```

**Comment by dschaper** [comment]
It's not `sudo` that makes it work since you're already the root user. The root user path is not correct.

Try looking at the location of `usermod` with `which usermod` or `where usermod` and not the full path. Then check the path as the root user `echo $PATH` and you'll likely see it doesn't include `usermod`. Running `sudo` as you did creates a new session that likely does include the path, `sudo echo $PATH` to check.

We suggest you don't use the root user to install Pi-hole, issues like this will happen when you do that.

>Call to undefined function json_encode()

The PHP JSON package didn't install.

**Comment by gonzague** [comment]
Thanks again. I removed using pihole uninstall and reinstalled... under another user than root ..

Same problem, with the web interface error. The system returns :
`php7.3-json is already the newest version (7.3.14-1~deb10u1).
php7.3-json set to manually installed.`

I ran `sudo phpenmod json` but don't know why .. when I do ..
`php -m
[PHP Modules]
Core
date
filter
hash
libxml
openssl
pcntl
pcre
Reflection
session
sodium
SPL
standard
zlib

[Zend Modules]
`

... no trace of JSON. 

Again on a fresh install of Debian 10 ; with nothing else installed.

**Comment by ctbfourone** [comment]
@gonzague Try checking your `$PATH`. On Debian 10.4, as root, my path was not inclusive enough. When I set it to `export PATH=/bin:/usr/bin:/usr/local/bin:/sbin:/usr/sbin:/usr/local/sbin:$PATH` so that I got all the bin locations and whatever else it already had set, then installation worked just fine.

Probably also make sure that apache2 is stopped, if installed, or remove it if you don't plan to use it, since both it and lighttpd will try to bind to port 80.
